packages/server/graphql/types/GitLabIntegration.ts

```diff
@@ -33,13 +33,13 @@ const GitLabIntegration = new GraphQLObjectType<any, GQLContext>({
       type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(IntegrationProviderOAuth2))),
       resolve: async ({userId}, _args, {dataLoader}) => {
         const teamMembers = await dataLoader.get('teamMembersByUserId').load(userId)
+        const organizationUsers = await dataLoader.get('organizationUsersByUserId').load(userId)
+        const orgIds = organizationUsers.map(({orgId}) => orgId)
```

Comment:
-1 As stated in line 9 and line 32 the integration is per team member, thus we don't want all organizations, but only the one where the current team is in.

Reply:
Could you clarify the difference between the `cloudProvider` and `sharedProviders` please? Am I right in saying that the `cloudProvider` is the default GitLab application (or Mattermost etc) that everyone can access, whereas the `sharedProviders` are the applications that a team or org has integrated with and only they can access? I'm a little unclear on this (see [comment here](https://github.com/ParabolInc/parabol/pull/5973#discussion_r794453361)). 

On line 35, we're getting all `teamMembers`, including ones that don't belong to this org or team. If we only want the providers integrated with this team or its org, I could do something like this:

```
        const viewerId = getUserId(authToken)
        const team = await dataLoader.get('teams').load(teamId)
        if (!team) return standardError(new Error('Team not found'), {userId: viewerId})
        const {orgId} = team
        const orgTeams = await dataLoader.get('teamsByOrgIds').load(orgId)
        const orgTeamIds = orgTeams.map(({id}) => id)
        return dataLoader
          .get('sharedIntegrationProviders')
          .load({service: 'gitlab', orgTeamIds, teamIds: [teamId]})
```

Reply:
You're right about `cloudProvider` and `sharedProviders` and your solution looks correct as well.

Reply:
OK thanks, update pushed 👍